ethcore/src/client/client.rs

```diff
@@ -276,6 +278,25 @@ impl Client {
 		self.notify.write().push(Arc::downgrade(&target));
 	}
 
+	/// Set a closure to call when we want to restart the client
+	pub fn set_exit_handler<F>(&self, f: F) where F: Fn(bool, Option<String>) + 'static + Send {
+		*self.exit_handler.lock() = Some(Box::new(f));
+	}
+
+	/// Hypervisor should restart this client.
+	pub fn restart(&self) {
+		if let Some(ref h) = *self.exit_handler.lock() {
+			(*h)(true, None);
+		}
+	}
+
+	/// Hypervisor should kill this client.
+	pub fn exit(&self) {
```

Comment:
i guess we might want to look into an alternative "hypervisor" IPC endpoint that could host such calls, however in the context of the present PR that would be too much extra baggage.